xsimlab/tests/test_xr_accessor.py

```diff
@@ -141,19 +175,43 @@ def test_clock_sizes(self):
 
         assert ds.xsimlab.clock_sizes == {"clock1": 3, "clock2": 2}
 
-    def test_master_clock_dim(self):
-        attrs = {self._clock_key: 1, self._master_clock_key: 1}
+    def test_main_clock_dim(self):
+        attrs = {self._clock_key: 1, self._main_clock_key: 1}
+        ds = xr.Dataset(coords={"clock": ("clock", [1, 2], attrs)})
+
+        assert ds.xsimlab.main_clock_dim == "clock"
+        assert ds.xsimlab._main_clock_dim == "clock"  # cache
+        assert ds.xsimlab.main_clock_dim == "clock"  # get cached value
+
+        ds = xr.Dataset()
+        assert ds.xsimlab.main_clock_dim is None
+
+    def test_master_clock_dim_warning(self):
+        attrs = {self._clock_key: 1, self._main_clock_key: 1}
         ds = xr.Dataset(coords={"clock": ("clock", [1, 2], attrs)})
 
-        assert ds.xsimlab.master_clock_dim == "clock"
-        assert ds.xsimlab._master_clock_dim == "clock"  # cache
-        assert ds.xsimlab.master_clock_dim == "clock"  # get cached value
+        with pytest.warns(
+            FutureWarning,
+            match="master_clock is to be deprecated in favour of main_clock",
+        ):
+            assert ds.xsimlab.master_clock_dim == "clock"
```

Comment:
```suggestion
            ds.xsimlab.master_clock_dim
```

nit: we're just testing the warning, so I think this test is enough (keeping the other `warns` tests below is fine, though).

Reply:
again `master_clock_dim` should still be working _with_ a warning, so why not test both? that way, it will also show when `master_clock_dim->main_clock_dim` linking is broken